From 6cab71f422f2425b4d2283023c9e955f9663dde6 Mon Sep 17 00:00:00 2001
From: Donald Stufft <donald@stufft.io>
Date: Sat, 20 Dec 2014 15:50:25 -0500
Subject: [PATCH] Fix ``pip list`` when there is an invalid PEP 440 version
 installed

---
 pip/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pip/__init__.py b/pip/__init__.py
index de256acd4..e3f6b4e70 100755
--- a/pip/__init__.py
+++ b/pip/__init__.py
@@ -253,7 +253,7 @@ def from_dist(cls, dist, dependency_links, find_tags=False):
             editable = False
             req = dist.as_requirement()
             specs = req.specs
-            assert len(specs) == 1 and specs[0][0] == '=='
+            assert len(specs) == 1 and specs[0][0] in ["==", "==="]
             version = specs[0][1]
             ver_match = cls._rev_re.search(version)
             date_match = cls._date_re.search(version)
